### PR TITLE
Feature/hide content

### LINKED ISSUE #100: As a gamer, I don't want to see pages that I don't have access to, so that I can understand more clearly what i can do
- [x] Remove pages from menu the user can't access
- [x] hide functionality the user shouldn't be used

### PR DESCRIPTION
**Pull request by Supergamer1337**
## Key Features
- Redirects the user if they try to access pages they don't have access to.
- Hides content from (home page) that users shouldn't be able to access.
- Refactored Remove Game route, because I couldn't stop myself.
- Closes #100 

## Known Issues
- Trying to delete a game that has ratings, borrows (that are not active) and playstatuses cannot be done. We could update the schema but that could have potential drawbacks. Similiar with just deleting the content before deleting the game. (This issue was there before this PR.)

### CODE REVIEW

**Comment by GAsplund** [comment]
> Trying to delete a game that has ratings, borrows (that are not active) and playstatuses cannot be done.

I only see two solutions to this problem:
1. Mark games as deleted in the database
2. Perform a cascading deletion

The first one would be ideal for keeping as much data as possible, but it makes it possible for other entities to refer to non-existent games. Unique constraints won't work either.
The second one would be good for having clean data, but if something was deleted in error, well oops, now literally all info you have on the game is gone.

Personally, I would go with the second option and making sure that the user really knows what they're doing.

**Comment by Supergamer1337** [comment]
> > Trying to delete a game that has ratings, borrows (that are not active) and playstatuses cannot be done.
> 
> I only see two solutions to this problem:
> 
> 1. Mark games as deleted in the database
> 2. Perform a cascading deletion
> 
> The first one would be ideal for keeping as much data as possible, but it makes it possible for other entities to refer to non-existent games. Unique constraints won't work either. The second one would be good for having clean data, but if something was deleted in error, well oops, now literally all info you have on the game is gone.
> 
> Personally, I would go with the second option and making sure that the user really knows what they're doing.

Yeah, that was my stance as well, but didn't want to make an executive decision on this without at least referring to rest of the team, since it does raise security concerns and affect large parts of the application.

File: frontend/src/pages/add.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { Inter } from 'next/font/google';
 import Head from 'next/head';
 import AddGame from '../components/AddGame/AddGame';
 import Header from '../components/Header/Header';
+import { redirectIfNotLoggedIn } from '../utils/loginRedirects';
 
-const inter = Inter({ subsets: ['latin'] });
```
Comment (x183) [inline frontend/src/pages/add.tsx:6]: Why is this removed?

Comment (Supergamer1337) [reply frontend/src/pages/add.tsx:6]: Unused

File: frontend/src/pages/admin/index.tsx [hunk 2]
```diff
@@ -1,10 +1,13 @@
-import { Inter } from 'next/font/google';
-import Head from 'next/head';
 import Header from '@/src/components/Header/Header';
 import styles from '@/src/pages/index.module.css';
+import { redirectIfNotAdmin } from '@/src/utils/loginRedirects';
+import { Inter } from 'next/font/google';
+import Head from 'next/head';
 
 const inter = Inter({ subsets: ['latin'] });
```
Comment (x183) [inline frontend/src/pages/admin/index.tsx:7]:
```suggestion
import Head from 'next/head';

```

**Review by x183 — APPROVED** [review]
Works fine :+1:
image
image

**Review by Oscariremma — APPROVED** [review]
LGTM
